alica_tests/src/test/test_config_change.cpp

```diff
@@ -0,0 +1,153 @@
+#include <gtest/gtest.h>
+
+#include "test_alica.h"
+
+#include "engine/planselector/PartialAssignment.h"
+namespace alica
+{
+    class AlicaEngine;
+    class AlicaContext;
+
+    TEST(ConfigUpdates, TestUpdatingComponents)
+{
+    ros::NodeHandle nh;
+    std::string path;
+    nh.param<std::string>("/rootPath", path, ".");
+
+    AlicaContext *ac = new alica::AlicaContext("RoleSet", "MasterPlan", true, path + "/etc/Alica_nase.yaml");
+    ac->setLocalAgentName("nase");
+
+    ASSERT_TRUE(ac->isValid());
+    ac->setCommunicator<alicaDummyProxy::AlicaDummyCommunication>();
+    alica::AlicaCreators creators(std::make_unique<alica::ConditionCreator>(), std::make_unique<alica::UtilityFunctionCreator>(),
+                                  std::make_unique<alica::ConstraintCreator>(), std::make_unique<alica::BehaviourCreator>());
+    AlicaEngine* ae = AlicaTestsEngineGetter::getEngine(ac);
+    const_cast<IAlicaCommunication&>(ae->getCommunicator()).startCommunication();
```

Comment:
creators initialization and communicator assignment and initialization can be done close to their place of use. i.e. before the init() call.